### PR TITLE
Move [-l, --list] option to "bundle-list" sub-command

### PR DESCRIPTION
**Pull request by MarioCarrilloA**
This group of patches allow that bundle-list gets [-l, --list] option from "bundle-add", furthermore this option is renamed as [-a, --all] in order to keep coherence and semantic.

### CODE REVIEW

File: docs/swupd.1 [hunk 1]
```diff
@@ -131,7 +125,14 @@ the listed bundles, either directly or indirectly\.
...
 
-List all installed sofware bundles in the local system\.
+List all installed sofware bundles in the local system\. Bundles available
+can be listed with the `\-\-all` option\.
+
+*  `\-a, \-\-all`
+
+    Lists all available software bundles, either installed or not, that
+    are available\. It will return 0 with succeeded and a different value
+    of 0 with failed.
```
Comment (fandeli) [inline docs/swupd.1]: This is a great step in documenting functionality specifying the cases when return is success or failure and the meaning, this should be rather be propagated to the rest of sub-commands man pages.

I also rather prefer to put an specific section in man page called "EXIT STATUS" to provide this information

File: test/functional/bundlelist/all/list/test.bats [hunk 2]
```diff
@@ -13,10 +13,8 @@ teardown() {
   clean_tars 10
 }
 
-@test "bundle-add list bundles" {
-  run sudo sh -c "$SWUPD bundle-add $SWUPD_OPTS --list"
-
-  check_lines "$output"
+@test "bundle-list all bundles" {
+  run sudo sh -c "$SWUPD bundle-add --all"
```
Comment (phmccarty) [inline test/functional/bundlelist/all/list/test.bats]: Can you restore the `check_lines` call? It's useful to verify swupd's expected printed output to actual output.

File: Makefile.am [hunk 3]
```diff
@@ -145,8 +145,8 @@ dist_check_SCRIPTS = \
 	test/functional/bundleadd/boot-file/test.bats \
 	test/functional/bundleadd/include/test.bats \
 	test/functional/bundleadd/fix-missing-file/test.bats \
-	test/functional/bundleadd/list/test.bats \
 	test/functional/bundleadd/verify-fix-path/test.bats \
+        test/functional/bundlelist/all/test.bats \
```
Comment (phmccarty) [inline Makefile.am]: this line should indented with a tab

**Comment by MarioCarrilloA** [comment]
@phmccarty done :) thanks

**Comment by tmarcu** [comment]
Can you rebase this to latest master?

**Comment by tmarcu** [comment]
+1 for updating tests too.

**Pull request merged**